stl/inc/xutility

```diff
@@ -3330,7 +3330,10 @@ namespace ranges {
         // clang-format off
         subrange() requires default_initializable<_It> = default;
 
-#if !defined(__clang__) && !defined(__EDG__) // TRANSITION, GH-2326 (originally DevCom-1331017)
+#if !defined(MSVC_INTERNAL_TESTING) && !defined(__clang__) && !defined(__EDG__) // TRANSITION, VS 17.1p3

```

Comment:
`MSVC_INTERNAL_TESTING` wasn't intended to be inspected by product code, which is why I didn't make it `_Ugly`. That said, I have no real concerns with shipping this briefly, and I've recorded the workaround for removal during the 17.1 Preview 3 toolset update.

Reply:
Oh, yuck. I suppose we can survive shipping this for one preview release.

Reply:
If you want, I could easily update the internal test harness to define both `MSVC_INTERNAL_TESTING` and `_Ugly` `_MSVC_INTERNAL_TESTING` when merging this, if you want to use the underscore-prefix in product code. Probably worthwhile on its own merits since it's been a point of confusion before. (`_ENABLE_STL_INTERNAL_CHECK` is already defined by our tests, but that doesn't say the right thing here, so I wouldn't recommend it.)

Reply:
How about if we change all occurrences of `MSVC_INTERNAL_TESTING` to `_MSVC_INTERNAL_TESTING` in this PR, and switch the internal definition when porting (instead of adding another define)? I'd prefer having just one name to two.

Reply:
Sounds good to me.

Reply:
There's the change here - we need to remember to change the one occurrence in `distrib.yml` when we merge.